### PR TITLE
Cleanup UnsafeRow related code structure

### PR DESCRIPTION
**Pull request by tanjialiang**
Major code structure refactoring and unused code path/tests cleanup. Currently it is often unclear and confusing looking at scattered classes and files. After the change it will be much easier to maintain.

* Migrated UnsafeRow deserialization from UnsafeRowDeserializer to UnsafeRowBatchDeserializer.
* Deprecated and removed UnsafeRowDeserializer.h and its corresponding tests.
* Renamed UnsafeRowBatchDeserializer.h to UnsafeRowDeserializers.h
* Combined UnsafeRowSerializer.h and UnsafeRowDynamicSerializer.h to a single file called UnsafeRowSerializers.h
* Moved UnsafeRow24Deserializer.* to a separate experimental directory for less confusion as it is experimental and not used in the main code path.
* Combined UnsafeRowDeserializerTest.cpp and UnsafeRowSerializerTest.cpp to a single file UnsafeRowSerdeTest.cpp
* Refactored tests accordingly.

### CODE REVIEW

File: velox/row/tests/UnsafeRowBatchDeserializerTest.cpp [hunk 1]
```diff
@@ -668,26 +667,33 @@ class UnsafeRowComplexBatchDeserializerTests
         {intVector, stringVector, intArrayVector, stringArrayVector}, isNullAt);
   }
 
+  void testVectorSerde(const RowVectorPtr& inputVector) {
```
Comment (mbasmanova) [inline velox/row/tests/UnsafeRowBatchDeserializerTest.cpp:670]: Can we consolidate all tests in velox/serializers/tests/UnsafeRowSerializerTest.cpp ?

File: velox/row/tests/UnsafeRowBatchDeserializerTest.cpp [hunk 2]
```diff
@@ -696,13 +702,7 @@ TEST_F(UnsafeRowComplexBatchDeserializerTests, nullRows) {
   for (auto& batchSize : {1, 5, 10}) {
     std::vector<std::optional<std::string_view>> serializedVector;
```
Comment (mbasmanova) [inline velox/row/tests/UnsafeRowBatchDeserializerTest.cpp:703]: serializedVector is not used anymore? Let's remove.

File: velox/row/tests/UnsafeRowBatchDeserializerTest.cpp [hunk 3]
```diff
@@ -716,17 +716,6 @@ TEST_F(UnsafeRowComplexBatchDeserializerTests, nullRows) {
...
-
-    auto rowSize = UnsafeRowDynamicSerializer::serialize(
-        outerRowVector->type(), outerRowVector, buffers_[0], 0);
-    ASSERT_TRUE(rowSize.has_value());
-    serializedVector.push_back(std::string_view(buffers_[0], rowSize.value()));
-    VectorPtr outputVector =
-        UnsafeRowDynamicVectorBatchDeserializer::deserializeComplex(
-            serializedVector, outerRowVector->type(), pool_.get());
-    assertEqualVectors(outerRowVector, outputVector);
+    testVectorSerde(outerRowVector);
```
Comment (mbasmanova) [inline velox/row/tests/UnsafeRowBatchDeserializerTest.cpp:719]: Is the TODO above still valid?

```
// TODO: The serde is still buggy as tests with innerBatchSize larger than 1
  // is currently still failing.
```

File: velox/row/tests/UnsafeRowBatchDeserializerTest.cpp [hunk 4]
```diff
@@ -716,17 +716,6 @@ TEST_F(UnsafeRowComplexBatchDeserializerTests, nullRows) {
     const auto& innerRowVector =
         createInputRow(innerBatchSize, [](vector_size_t i) { return true; });
     const auto& outerRowVector = makeRowVector({innerRowVector});
```
Comment (mbasmanova) [inline velox/row/tests/UnsafeRowBatchDeserializerTest.cpp:718]: It seems wrong to store the results of makeRowVector in a const &.

**Review by mbasmanova — APPROVED** [review]
@tanjialiang Thank you for the cleanup.

File: velox/row/tests/UnsafeRowBatchDeserializerTest.cpp (hunk 3 above)
Comment (tanjialiang) [reply velox/row/tests/UnsafeRowBatchDeserializerTest.cpp:719]: Yeah it is still valid. This is the bug we need to fix after refactoring

**Comment by vermapratyush** [comment]
@tanjialiang can you extract out the migration of UnsafeRowDeserializer to UnsafeRowBatchDeserializer in a separate PR as it is a behavioural change. It would be good to test out the change independently.

**Pull request closed without merging**